vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/patch_flags.go

```diff
@@ -90,32 +92,25 @@ func applyOpenShiftConfigControllerArgs(controllerManagerOptions *options.KubeCo
 			args[key] = append(args[key], arrayValue.(string))
 		}
 	}
-	if err := applyFlags(args, controllerManagerOptions.Flags(KnownControllers(), ControllersDisabledByDefault.List())); len(err) > 0 {
+	if err := applyFlags(args, cmd.Flags()); len(err) > 0 {
 		return kerrors.NewAggregate(err)
 	}
 	return nil
 }
 
 // applyFlags stores the provided arguments onto a flag set, reporting any errors
 // encountered during the process.
-func applyFlags(args map[string][]string, flags flag.NamedFlagSets) []error {
+func applyFlags(args map[string][]string, flags *pflag.FlagSet) []error {
```

Comment:
why is all this still needed?

Reply:
Since we wire and pass these flag values, we can't break them. I don't want to end up in the situation where this stops working, although truth to be told, this never worked in past few releases :sweat: In 4.6 this will be entirely dropped in favor of regular flag passing.